src/librustc_back/svh.rs

```diff
@@ -51,19 +51,26 @@ use std::hash::{Hash, SipHasher, Hasher};
 use syntax::ast;
 use syntax::visit;
 
-#[derive(Clone, PartialEq, Debug)]
+#[derive(Clone, PartialEq, Eq, Debug)]
 pub struct Svh {
-    hash: String,
+    raw: u64,
 }
 
 impl Svh {
     pub fn new(hash: &str) -> Svh {
         assert!(hash.len() == 16);
-        Svh { hash: hash.to_string() }
-    }
+        // Ideally we'd just reverse the nibbles on LE machines during to_string, unfortunately
+        // this would break the abi so I guess we're just doing this now.
+
+        let s = if cfg!(target_endian = "big") {
+            hash.to_string()
+        } else {
+            hash.chars().rev().collect()
+        };
```

Comment:
In the to_string() impl, it walks the low nibble to the high nibble constructing the string, the net result being that without this, `Svh::new("deadbeefdeadbeef").to_string()` returns `feebdaedfeebdaed`

Unfortunately, because it's flipped nibble wise not byte wise I can't just flip the bytes in the resulting u64, but on a big endian architecture this reversing shouldn't be necessary.